src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -0,0 +1,224 @@
+
+        /// <summary>
+        /// Creates an ODataRequestMessage for a bulk update request.
+        /// </summary>
+        /// <returns>Returns an instance of the ODataRequestMessage for the bulk update request.</returns>
+        private ODataRequestMessageWrapper CreateBulkUpdateRequest(BulkUpdateGraph bulkUpdateGraph)
+        {
+            EntityDescriptor baseDescriptor = bulkUpdateGraph.TopLevelDescriptors[0] as EntityDescriptor;
+            Uri requestUri = UriUtil.CreateUri(this.RequestInfo.BaseUriResolver.GetBaseUriWithSlash(), UriUtil.CreateUri(bulkUpdateGraph.EntitySetName, UriKind.Relative));
+            Version requestVersion = Util.ODataVersion401;
+            string httpMethod = this.GetHttpMethod(EntityStates.Modified, ref requestVersion);
+
+            HeaderCollection headers = new HeaderCollection();
+
+            this.RequestInfo.Context.Format.SetRequestContentTypeForEntry(headers);
+
+            ApplyPreferences(headers, httpMethod, this.RequestInfo.AddAndUpdateResponsePreference, ref requestVersion);
+
+            headers.SetHeader("OData-Version", "4.01");
+            headers.SetHeader("OData-MaxVersion", "4.01");
+
+            this.RequestInfo.Format.SetRequestAcceptHeader(headers);
+
+            return this.CreateRequestMessage(httpMethod, requestUri, headers, this.RequestInfo.HttpStack, baseDescriptor, this.IsBatchRequest ? baseDescriptor.ChangeOrder.ToString(CultureInfo.InvariantCulture) : null);
+        }
+
+        /// <summary>
+        /// Generate the bulk update request for all changes to save.
+        /// </summary>
+        /// <returns>Returns the instance of ODataRequestMessage containing all the headers and payload for the delta request.</returns>
+        private ODataRequestMessageWrapper GenerateBulkUpdateRequest()
+        {
+            if (!this.bulkUpdateGraph.TopLevelDescriptors.Any())
+            {
+                this.SetCompleted();
+                return null;
+            }
+
+            ODataRequestMessageWrapper bulkUpdateRequestMessage = this.CreateBulkUpdateRequest(this.bulkUpdateGraph);
+
+            // we need to fire request after the headers have been written, but before we write the payload
+            bulkUpdateRequestMessage.FireSendingRequest2(descriptor: null, true);
+
+            using (ODataMessageWriter messageWriter = Serializer.CreateDeltaMessageWriter(bulkUpdateRequestMessage, this.RequestInfo, isParameterPayload: false))
+            {
+                this.writer = ODataWriterWrapper.CreateForDeltaFeed(messageWriter, this.bulkUpdateGraph.EntitySetName, this.RequestInfo.Configurations.RequestPipeline, bulkUpdateRequestMessage, this.RequestInfo);
+                this.SerializerInstance.WriteDeltaResourceSet(this.bulkUpdateGraph.TopLevelDescriptors, this.bulkUpdateGraph, writer);
+            }
+
+            return bulkUpdateRequestMessage;
+        }
+
+        protected override bool ProcessResponsePayload => throw new NotImplementedException();

```

Comment:
Give a description about why "not implemented"?

Reply:
This will be implemented with the bulk update reading changes.